pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -801,91 +800,53 @@ func createHostDnsmasqFile(ctx *zedrouterContext, bridge string) {
 	}
 }
 
-// Returns an IP address as a string, or "" if not found.
-func lookupOrAllocateIPv4(
-	ctx *zedrouterContext,
-	status *types.NetworkInstanceStatus,
-	mac net.HardwareAddr) (string, error) {
+// Returns an IP address as a string, or "" on error
+func lookupOrAllocateIPv4(status *types.NetworkInstanceStatus,
+	appID uuid.UUID, appNum int, mac net.HardwareAddr) (string, error) {
+	log.Functionf("lookupOrAllocateIPv4(%s-%s): appNum:%d\n",
+		status.DisplayName, status.Key(), appNum)
 
-	log.Functionf("lookupOrAllocateIPv4(%s-%s): mac:%s\n",
-		status.DisplayName, status.Key(), mac.String())
-	// Lookup to see if it exists
+	networkID := status.UUIDandVersion.UUID
+	// if it exists
 	if ip, ok := status.IPAssignments[mac.String()]; ok {
-		log.Functionf("found Ip addr ( %s) for mac(%s)\n",
-			ip.String(), mac.String())
+		log.Functionf("lookupOrAllocateIPv4(%s-%s): found %s for %s\n",
+			networkID.String(), appID.String(), ip.String(), mac.String())
 		return ip.String(), nil
 	}
-
-	log.Functionf("bridgeName %s Subnet %v range %v-%v\n",
-		status.BridgeName, status.Subnet,
-		status.DhcpRange.Start.String(), status.DhcpRange.End.String())
-
 	if status.DhcpRange.Start == nil {
 		if status.Type == types.NetworkInstanceTypeSwitch {
-			log.Functionf("%s-%s switch means no bridgeIpAddr",
+			log.Functionf("%s-%s switch means no IPAddr",
 				status.DisplayName, status.Key())
 			return "", nil
 		}
 		log.Fatalf("%s-%s: nil DhcpRange.Start",
```

Comment:
Seems like the return nil in parseIpspec() when addrCount == 0 can result in later hitting this fatal.
So really need to understand why you added that check and make it not return a failure.

Reply:
seems like the caller checks for nil, so should never hit this fatal.